src/common/utils/use-waiting-message.ts

```diff
@@ -0,0 +1,33 @@
+import { useRef, useState } from 'react';
+import { useInterval } from './use-interval';
+
+// Keys are the seconds to wait before showing the message
+export type WaitingMessages = Record<number, string>;
+
+function messageForSecondsPassed(waitingMessages: WaitingMessages, seconds: number) {
+  return waitingMessages[seconds as keyof typeof waitingMessages];
+}
+
+export const useWaitingMessage = (
+  waitingMessages: WaitingMessages,
+  { waitingMessageInterval } = {
+    waitingMessageInterval: 1000,
+  }
+): [string, () => void, () => void] => {
+  const [isRunning, setIsRunning] = useState(false);
+  const [waitingMessage, setWaitingMessage] = useState(messageForSecondsPassed(waitingMessages, 0));
+  const stopWaitingMessage = () => setIsRunning(false);
+  const startWaitingMessage = () => setIsRunning(true);
```

Comment:
Same issue as before. The fact these these declared without `useMemo` and they're used in the `useCallback` in `unlock.tsx`, that fn is redeclared on every rerender. Could also use `useCallback`.

```suggestion
  const handlers = useMemo(() => ({
    stopWaitingMessage: () => setIsRunning(false),
    startWaitingMessage: () => setIsRunning(true),
  }), []);
```